### PR TITLE
[doc] add docs for retool workflow

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Added developer docs for Retool workflow

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Checkout the branch, install dependencies with `npm ci`, then run `npm run docs:dev` and navigate to `localhost:8080` to review.

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @varortz

### CODE REVIEW

File: docs/dev/retool/workflow.md [hunk 1]
```diff
@@ -1 +1,74 @@
...
+## Legal Server Sync
+
+To get data from Legal Server, we added the provided API link as a REST API resource.
+After fetching data from Legal Server using said resource, we processed it as follows:
+
+1. Filter to only open cases
+2. Search legal problem code for keywords (i.e. asylum)
+3. Determine time commitment by legal problems
+4. Upsert cases and relevant languages and relief codes to Supabase
+5. Delete old stored cases that do not match what's stored in Legal Server
```
Comment (varortz) [inline docs/dev/retool/workflow.md]: gonna drop some wording nits:
- Can we update 1 to: "Filter to only cases seeking volunteers"
- Can we update 3 to: "Determine time commitment based on the case's relief sought value and whether the client is detained"
- 5: should be "LegalServer"

File: docs/dev/retool/workflow.md [hunk 2]
```diff
@@ -1 +1,74 @@
 # Retool Workflow
+
+[Retool workflows](https://retool.com/products/workflows) are great for automated tasks that need to be run on intervals. For our project, we use Retool workflows to routinely sync our database with Legal Server every day. Additionally, we take advantage of Retool workflows to create function abstractions that chain REST APIs to create, edit, or delete listings (i.e. limited case assignments) via the Retool application.
+
+## Legal Server Sync
+
+To get data from Legal Server, we added the provided API link as a REST API resource.
+After fetching data from Legal Server using said resource, we processed it as follows:
```
Comment (varortz) [inline docs/dev/retool/workflow.md]: nit: "LegalServer" is one word

File: docs/dev/retool/workflow.md [hunk 3]
```diff
@@ -1 +1,74 @@
...
+
+1. Filter to only open cases
+2. Search legal problem code for keywords (i.e. asylum)
+3. Determine time commitment by legal problems
+4. Upsert cases and relevant languages and relief codes to Supabase
+5. Delete old stored cases that do not match what's stored in Legal Server
+
```
Comment (varortz) [inline docs/dev/retool/workflow.md:96]: Could we add another section titled "Other Workflows" that lists the other workflows? It might also be helpful to add a brief description of what each workflow achieves, doesn't need to be as detailed as the LegalServer sync workflow above.

image

File: docs/dev/retool/workflow.md [hunk 4]
```diff
@@ -1 +1,74 @@
 # Retool Workflow
+
+[Retool workflows](https://retool.com/products/workflows) are great for automated tasks that need to be run on intervals. For our project, we use Retool workflows to routinely sync our database with Legal Server every day. Additionally, we take advantage of Retool workflows to create function abstractions that chain REST APIs to create, edit, or delete listings (i.e. limited case assignments) via the Retool application.
```
Comment (varortz) [inline docs/dev/retool/workflow.md]: let's update "Legal Server" to "LegalServer"

File: docs/dev/retool/workflow.md [hunk 5]
```diff
@@ -1 +1,74 @@
...
+
+## Legal Server Sync
```
Comment (varortz) [inline docs/dev/retool/workflow.md]: should be "LegalServer", but also, thoughts on renaming this section to "LegalServer Sync Workflow"?

File: docs/dev/retool/workflow.md [hunk 6]
```diff
@@ -1 +1,74 @@
...
+5. Delete old stored cases that do not match what's stored in Legal Server
+
+## Supabase REST API
+
+::: warning
+The Supabase REST API resource on Retool (titled IJP Supabase Service) uses the service key, which can bypass row level protection. NEVER expose this directly to the user.
+:::
+
+We use this API to make requests to Supabase such as inserting, deleting, and modifying data.
+For example, in syncing to Legal Server, we need to use this resource to upsert (insert, or update if it already exists) the cases data that we fetched from Legal Server.
```
Comment (varortz) [inline docs/dev/retool/workflow.md]: nit: "LegalServer" here as well 😔

File: docs/dev/retool/workflow.md [hunk 7]
```diff
@@ -1 +1,74 @@
...
+To specify additional columns to detect conflicts on, add a URL query parameter `on_conflict`.
+For example:
+
+```text:no-line-numbers
+URL: .../v1/<table_name>?on_conflict=<column_name>
+```
+
+![Example HTTP request to perform an UPSERT operation](/assets/image/example_upsert_operation.png)
+
+### DELETE
```
Comment (varortz) [inline docs/dev/retool/workflow.md:140]: wowoow! Thanks for adding these sections :D

**Review by varortz — COMMENTED** [review]
LOOKING GOOD JINKANG! Should be good to merge after addressing the some nits + edits.
